### PR TITLE
Implementing matrix inversion

### PR DESCRIPTION
**Pull request by takuseno**
I've implemented `batch_inv` on GPU extension.

https://github.com/sony/nnabla/pull/351

### CODE REVIEW

**Comment by takuseno** [comment]
I've confirmed the unit test has passed with the same test at sony/nnabla#351 by enabling cuda extension.

File: src/nbla/cuda/function/generic/batch_inv.cu [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+template <typename T>
+void BatchInvCuda<T>::forward_impl(const Variables &inputs,
+                                      const Variables &outputs) {
+  cuda_set_device(this->device_);
+  const Tc *x = inputs[0]->get_data_pointer<Tc>(this->ctx_);
+  Tc *y = outputs[0]->cast_data_and_get_pointer<Tc>(this->ctx_, true);
+
+  shared_ptr<CudaCachedArray> pivot =
+    make_shared<CudaCachedArray>(dim_ * batch_size_, dtypes::INT, this->ctx_);
+  pivot->zero();
```
Comment (TE-TakuyaNarihira) [inline src/nbla/cuda/function/generic/batch_inv.cu]: Is this `zero()` required? I couldn't find that from the official document.

File: src/nbla/cuda/function/generic/batch_inv.cu [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+  shared_ptr<CudaCachedArray> info =
+    make_shared<CudaCachedArray>(batch_size_, dtypes::INT, this->ctx_);
+  info->zero();
```
Comment (TE-TakuyaNarihira) [inline src/nbla/cuda/function/generic/batch_inv.cu]: This as well.

File: src/nbla/cuda/function/generic/batch_inv.cu [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+template <typename T>
+void BatchInvCuda<T>::forward_impl(const Variables &inputs,
+                                      const Variables &outputs) {
+  cuda_set_device(this->device_);
+  const Tc *x = inputs[0]->get_data_pointer<Tc>(this->ctx_);
+  Tc *y = outputs[0]->cast_data_and_get_pointer<Tc>(this->ctx_, true);
+
+  shared_ptr<CudaCachedArray> pivot =
```
Comment (TE-TakuyaNarihira) [inline src/nbla/cuda/function/generic/batch_inv.cu]: You don't have to use shared pointers (I know it is used in other functions though...)

**Comment by takuseno** [comment]
@TE-TakuyaNarihira 
Hi! I've refactored my codes!

**Comment by TE-AkioHayakawa** [comment]
I've created a new PR to merge this branch. See #192.

**Pull request closed without merging**